### DISCUSSION TITLE
Quarkus at JNation 2024

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#40295] [Events]
**Event Description:** JNation is an inclusive developer conference covering various topics relevant to the software development industry.

Attendees have the unique opportunity to get up to speed with the latest and greatest technology trends in programming languages like Java and Javascript, Cloud & Infrastructures, Machine Learning and Artificial Intelligence, or Security and Methodologies.

We bring the actual technology experts working in these areas, so attendees have the chance to learn with the best!

**Date:** June 4-5, 2024
**Location:** Coimbra, Portugal
**Event Type:** In Person

https://jnation.pt/

**Session Title:** A day in the life of a Quarkus developer 
**Speaker(s):**  Martin Stefanko & Andy Damevin
**Day/Time:**  Tuesday | 3:00 pm - 3:50 pm
**Abstract:** If you are a Java developer, chances are you’ve already heard about Quarkus. You may have heard about the developer joy that Quarkus provides. But what exactly is it? How does it enhance your day-to-day tasks? What benefits and shortcuts does it offer for your work? These are the questions we will address in this talk.

During this live-coded session, we will create a new Quarkus project and demonstrate how it can expedite the development of Java microservices by leveraging its multitude of out-of-the-box features. The session will be interactive, so if there is something specific you want to see, we will gladly demonstrate it. Our primary focus, however, will be showcasing typical user application development scenarios, including database integration, remote connections using both blocking and asynchronous APIs, and implementing security measures. Naturally, we cannot overlook the crucial aspects of packaging applications into containers and deploying them to the cloud.

Quarkus is a framework that places developer productivity as a top priority, fostering a sense of joy throughout the development process. You will witness the remarkable things we can achieve with Quarkus within the given time. By the end of the session, you will truly grasp the essence of Quarkus' development joy.

**Session Title:** Building a JVM-Based Cloud Native API Gateway
**Speaker(s):**  Pedro Andrez
**Day/Time:** Tuesday | 12:10 pm - 1:00 pm
**Abstract:** The session will present how Mercedes-Benz.io developed an API Gateway that provides Internet-facing access to all services that compose our commerce solution. This solution was named Frontend Facing Gateway (FFG)
The session will focus on the technical aspects of the solution and also the tools used to interact with it from a user's perspective.
The session will cover the main pillars of the solution:
1 - The API Gateway itself, describing how custom filters are implemented and used and how the automatic route reloading was implemented
2 - The Kubernetes Operator as the main interaction tool to manage the routes exposed in the Gateway.

As a cloud native solution, the session will also cover Kubernetes topics (Mutating & Validating Webhooks and Operators) that were crucial to the overall solution.

The main tech stack for this solution is:
- Kotlin
- Spring Cloud Gateway
- Spring Boot
- Quarkus
- MongoDB
- Kubernetes

**Comment by xstefank** [comment]
A day in the life of a Quarkus developer is at 3:00 pm - 3:50 pm 

and Building a JVM-Based Cloud Native API Gateway is at 1:10 pm - 2:00 pm.

https://jnation.pt/schedule/.

**Reply by insectengine** [reply]
updated... thanks!

**Reply by xstefank** [reply]
@insectengine sorry, I didn't notice that the schedule was set to my timezone, so -1 h for Coimbra time.
